### DISCUSSION TITLE
A few questions about webhook security and multiple homes.

### DISCUSSION THREAD

**Discussion by kshartman** [iklein99/homebridge-smartthings#158] [Q&A]
First off, thanks for the great plugin.  I have a few questions about security.  I don't mind paying for access to a great feature but I need to know a few things so I can trust it.

1. Where is the webhook server hosted, I assume you are using AWS or some other secure cloud provider?

2.  I have multiple homes. Each home has its own smartthings account and each invites the smartthings account of my wife and myself to their home.  Can I use the same webhoook token in each home or do I need to purchase a webhook token for each? I see that you have a filter for home in the plugin which probably means I could install the webapp in emy smartthings account (which has access to each home) and use that filter in each homebridge server to control the scope, but that is not how I want to set it up. I am hoping I can create another smart app and use the same webhook token in each account.  If not, then buying a token for each home will certainly work, correct?

3. Even though you do have a device filter in the homebridge plugin, the smartthings app requests access to all devices (and then I presume you apply the filter in the webhook or in the plugin to exclude those devices).  This means that your code will have access to the devices I want to filter, such as locks, even though I won't see them in the interface, correct?  For now I am simply going to remove all locks and security devices from smartthings, which is kind of a pain.  What I would prefer is if ether a) I could run the webhook code on my own server and pay you an annual support and update fee or b) the code which implements the smart app would allow me to exclude devices (or specify which devices I want you to have) rather than requesting access to all devices.  Is either a) or b) possible?

Thanks!

**Comment by iklein99** [comment]
1. Hosted in AWS.  Access to the server is restricted publicly to port 80 (http) and 443(https).  Your password is encrypted (one-way).  The only info I capture is the number of events that are being generated.  I don't store information about your devices or locations on the server.
2. The smart app gets associated with your ST account.  If you have multiple accounts, you will need one set up for each account.  The token will associate your homebridge with the last app you install.  It doesn’t support association with multiple accounts.  The ignore locations will only help you out if you have one ST account.  I don’t know for sure what will happen if you try to use multiple locations with multiple instances of homebridge against the same smart app.  Never tested that.  My guess is that won’t work, but I’d have to look at it.
3. The way the service works is that while you are providing access to all devices, it will ignore any device that isn’t being managed by your homebridge plugin.  The plugin sends in a request for events and lists the devices it is interested in.  Any other devices are ignored.

**Reply by kshartman** [reply]
Thanks! Your app works really well. I will purchase another token for each home.

**Comment by iklein99** [comment]
Maybe let me think about this a little bit to see if I can come up with a better option for you.  Otherwise you will need to create a unique account with a unique email address for each location.  I only intended for each user to register once.

Would you consider consolidating your st hubs onto one smart things account?
